### PR TITLE
Reverse words in a String C++

### PR DESCRIPTION
**Pull request by HariniJeyaraman**
**Fixes issue:**

#4732

**Changes:**

Reverse words in a given string in C++

### CODE REVIEW

**Comment by HariniJeyaraman** [comment]
@arnavb Please review my code

**Comment by HariniJeyaraman** [comment]
@arnavb Please review the code for reversing words in a string

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+void reverseWords(std::string s) {
+    std::string temp;
+    int c = 1;
+    s.resize(s.length());
+    temp.resize(s.length());
+    for (int i = 0; i < s.length(); ++i) {
```
Comment (arnavb) [inline code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp]: Range based for loop.

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+#include <algorithm>
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+#include <vector>
+
+void reverseWords(std::string s) {
+    std::string temp;
+    int c = 1;
+    s.resize(s.length());
```
Comment (arnavb) [inline code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp]: What is the purpose of this line?

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        k++;
+        if ((isspace(s[i])) || i == s.length() - 1) {
+            str[ind] = temp;
+            temp.clear();
+            temp.resize(s.length());
+            ind++;
+            k = 0;
+        }
+    }
+    std::cout << "After Reversal of Words in String : \n";
```
Comment (arnavb) [inline code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp]:
```suggestion
```

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    std::reverse(str.begin(), str.end());
+    for (std::string s : str)
```
Comment (arnavb) [inline code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp]: Just return str and don't print anything in this function.

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp (hunk 2 above)
Comment (HariniJeyaraman) [reply code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp]: I was able to assign s to temp index by index only after resizing the strings

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp (hunk 4 above)
Comment (HariniJeyaraman) [reply code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp]: done

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp (hunk 1 above)
Comment (HariniJeyaraman) [reply code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp]: done

**Comment by HariniJeyaraman** [comment]
@arnavb Please review the changes

File: code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp (hunk 2 above)
Comment (arnavb) [reply code/string_algorithms/src/reverse_word_string/reverse_word_string.cpp]: @HariniJeyaraman I've tried out your code online and it [works fine](https://repl.it/repls/IllegalMoccasinAbstracttype) with this line commented out. Also, I would like that you remove the extra spaces between each of the words when printing. Some words have 2, others have 1 space in between.

**Comment by HariniJeyaraman** [comment]
@arnavb Sorry for the trouble. I think the updated code must be alright now. Please review the code

**Pull request merged**